src/btreemap/node.rs

```diff
@@ -49,10 +49,11 @@ pub type Entry<K> = (K, Vec<u8>);
 #[derive(Debug)]
 pub struct Node<K: Storable + Ord + Clone> {
     address: Address,
-    keys: Vec<K>,
+    // List of tuples consisting of a key and the encoded value.
     // Values are stored in a Refcell as they are loaded lazily.
     // A RefCell allows loading the value and caching it without requiring exterior mutability.
-    encoded_values: RefCell<Vec<Value>>,
+    // INVARIANT: the list is sorted by key.
+    keys_encoded_values: Vec<(K, RefCell<Value>)>,
```

Comment:
Perhaps `keys_and_encoded_values` is a bit better? It took me a moment there to parse the name (until I wrapped my head around the tuple).